remoting/remoting-triple/src/test/java/com/alipay/sofa/rpc/server/triple/UniqueIdInvokerTest.java

```diff
@@ -0,0 +1,122 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.server.triple;
+
+import com.alipay.sofa.rpc.config.ProviderConfig;
+import com.alipay.sofa.rpc.context.RpcInvokeContext;
+import com.alipay.sofa.rpc.core.exception.SofaRpcException;
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.invoke.Invoker;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class UniqueIdInvokerTest {
+
+    private static final String INTERFACE_ID    = "com.alipay.test.UniqueIdFacade";
+
+    private static final String UNIQUE_ID_ONE   = "unique-1";
+
+    private static final String UNIQUE_ID_TWO   = "unique-2";
+
+    private static final String UNIQUE_ID_THREE = "unique-3";
+
+    @Test(expected = SofaRpcException.class)
+    public void test() {
+        try {
+            // Prepare data
+            UniqueIdInvoker uniqueIdInvoker = new UniqueIdInvoker();
+
+            ProviderConfig uniqueOne = new ProviderConfig();
+            uniqueOne.setInterfaceId(INTERFACE_ID);
+            uniqueOne.setUniqueId(UNIQUE_ID_ONE);
+            uniqueIdInvoker.registerInvoker(uniqueOne, new UniqueIdTestInvoker(UNIQUE_ID_ONE));
+
+            ProviderConfig uniqueTwo = new ProviderConfig();
+            uniqueTwo.setInterfaceId(INTERFACE_ID);
+            uniqueTwo.setUniqueId(UNIQUE_ID_TWO);
+            uniqueIdInvoker.registerInvoker(uniqueTwo, new UniqueIdTestInvoker(UNIQUE_ID_TWO));
+
+            // Case 1: Invoke invoker which unique id is one
+            SofaRequest sofaRequest = new SofaRequest();
+            sofaRequest.setInterfaceName(INTERFACE_ID);
+            RpcInvokeContext.getContext().put(TripleContants.SOFA_UNIQUE_ID, UNIQUE_ID_ONE);
+            SofaResponse sofaResponse = uniqueIdInvoker.invoke(sofaRequest);
+            String appResponse = (String) sofaResponse.getAppResponse();
+            Assert.assertEquals(appResponse, UNIQUE_ID_ONE);
+
+            // Case 2: Invoke invoker which unique id is two
+            RpcInvokeContext.getContext().put(TripleContants.SOFA_UNIQUE_ID, UNIQUE_ID_TWO);
+            sofaResponse = uniqueIdInvoker.invoke(sofaRequest);
+            appResponse = (String) sofaResponse.getAppResponse();
+            Assert.assertEquals(appResponse, UNIQUE_ID_TWO);
+
+            // Case 3: There was only one invoker in UniqueIdInvoker, can not find invoker with unique id two,
```

Comment:
We need to discuss the reasonableness of case3, I think the case of uniqueId mismatch should not fallback to other uniqueId`s invoker.

Reply:
In the last implement of uniqueId for triple, a server can serve request with path  `/{interfaceId}/{method}` or '/{interfaceId}.{uniqueId}/{method}'.  A client with very old version of SOFARPC may send a request with path `/{interfaceId}/{method}`  and without uniqueId info in header. So if there is only one invoker and **uniqueId is empty**, we should fallback to the only invoker.

Reply:
I think  a check for if **uniqueid is empty** should added here.

```java
if(StringUtils.isEmpty(uniqueId)){
    // falback
}else{
    //  throw exception
}
```

Reply:
OK, i will fix this